src/app/tracing/data.model.ts

```diff
@@ -180,7 +180,9 @@ export interface ShowElementsTraceParams {
 
 export interface GraphSettings {
     type: GraphType;
-    mapType: MapType;
+    mapVariant: MapVariant;
```

Comment:
Please use 
```
  mapSettings: {
    tileServer: TileServer;
    mapType: MapType;
    shapeFileData: ShapeFileData | null;
    geojsonBorderWidth: number;
    geojsonBorderColor: Color;
  }
```
instead of 
```
  mapVariant: MapVariant;
  lastMapTypeSelected: MapType.MAPNIK;
  shapeFileData: ShapeFileData | null;
  geojsonBorderWidth: number;
  geojsonBorderColor: Color;
```

you could type the nested object as MapSetttings